features/matchers/have_http_status_matcher.feature

```diff
@@ -1,12 +1,16 @@
 Feature: `have_http_status` matcher
 
   The `have_http_status` matcher is used to specify that a response returns a
-  desired status code.
+  desired status code. It accepts one argument in any of the following formats:
+
+  * numeric
+  * name (defined in `Rack::Utils::SYMBOL_TO_STATUS_CODE`)
+  * type (`:success`, `:missing`, `:redirect`, or `:error`)
```

Comment:
Rails name rather than type...

Reply:
I'm going to consolidate the scenario-naming conversation with this one, since they address the same issue.

I went back to the source code for this matcher, and noticed two things:

1. The "Rails name" arguments are defined in the [`GenericStatus` class](https://github.com/rspec/rspec-rails/blob/master/lib/rspec/rails/matchers/have_http_status.rb#L239), in a [`type_codes`](https://github.com/rspec/rspec-rails/blob/master/lib/rspec/rails/matchers/have_http_status.rb#L315) method. Thus, I'm proposing the term "generic status type".
2. The four possible values for generic status types are defined explicitly in the aforementioned `type_codes` method. I can't find any other place that lists exactly those four possible values (the [`ActionDispatch::TestResponse` documentation](https://edgeapi.rubyonrails.org/classes/ActionDispatch/TestResponse.html) doesn't include `#redirect?`, and the `Rack::Response::Helpers` module that it mixes in [doesn't include `#missing?` and _does_ include many others](https://github.com/rack/rack/blob/master/lib/rack/response.rb#L113-L134)), so I'm proposing we just list them explicitly here.

```
* numeric code
* status name, as defined in `Rack::Utils::SYMBOL_TO_STATUS_CODE`
* generic status type (`:success`, `:missing`, `:redirect`, or `:error`)

Scenario: Checking a numeric status code
Scenario: Checking a symbolic status name
Scenario: Checking a symbolic generic status type
```